src/Catrobat/Controller/Web/NotificationsController.php

```diff
@@ -0,0 +1,216 @@
+      'instance' => $notification_instance,
+    ]);
+    $response->headers->set('Cache-Control', 'no-store, must-revalidate, max-age=0');
+    $response->headers->set('Pragma', 'no-cache');
+
+    return $response;
+  }
+
+  /**
+   * @Route("/user_notifications/markasread/{notification_id}", name="notification_mark_as_read",
+   * requirements={"notification_id": "\d+"}, defaults={"notification_id": null}, methods={"GET"})
+   */
+  public function markNotificationAsRead(int $notification_id,
+                                              CatroNotificationService $notification_service,
+                                              CatroNotificationRepository $notification_repo): JsonResponse
+  {
+    $user = $this->getUser();
+    if (null === $user)
+    {
+      return JsonResponse::create(['success' => false, 'message' => 'User not logged in']);
+    }
+    $notification_seen = $notification_repo->findOneBy(['id' => $notification_id, 'user' => $user]);
+    if (null === $notification_seen)
+    {
+      return new JsonResponse(['success' => false,
+        'message' => 'Notification not found or doesnt belong to user', ]);
+    }
+    $notification_service->markSeen([$notification_seen]);
+
+    return new JsonResponse(['success' => true]);
+  }
+
+  /**
+   * @Route("/user_notifications/notifications/count", name="notifications_count", methods={"GET"})
+   */
+  public function notificationsCountAction(CatroNotificationRepository $notification_repo,
+                                               RemixManager $remix_manager): JsonResponse
+  {
+    /** @var User|null $user */
+    $user = $this->getUser();
+    if (!$user)
+    {
+      return JsonResponse::create(['statusCode' => StatusCode::LOGIN_ERROR]);
+    }
+
+    $catro_user_notifications_all = $notification_repo->findBy(['user' => $user]);
+    $likes = 0;
+    $followers = 0;
+    $comments = 0;
+    $remixes = 0;
+    $all = 0;
+    foreach ($catro_user_notifications_all as $notification)
+    {
+      /** @var CatroNotification $notification */
+      if ($notification->getSeen())
+      {
+        continue;
+      }
+
+      if ($notification instanceof LikeNotification)
+      {
+        ++$likes;
+      }
+      elseif ($notification instanceof FollowNotification || $notification instanceof NewProgramNotification)
+      {
+        ++$followers;
+      }
+      elseif ($notification instanceof CommentNotification)
+      {
+        ++$comments;
+      }
+      elseif ($notification instanceof RemixNotification)
+      {
+        ++$remixes;
+      }
+
+      ++$all;
+    }
+
+    $unseen_remixed_program_data = $remix_manager->getUnseenRemixProgramsDataOfUser($user);
+
+    return new JsonResponse([
+      'count' => ['all-notifications' => $all,
+        'all-notifications-dropdown' => $all,
+        'likes' => $likes,
+        'followers' => $followers,
+        'comments' => $comments,
+        'remixes' => $remixes, ],
+      'statusCode' => 200,
```

Comment:
again the status code

+ the count is then not needed anymore. just use the (internal)  mapped array

```
['all-notifications' => $all,
 'all-notifications-dropdown' => $all,
 'likes' => $likes,....
```

Reply:
all-notification and all notification-dropdown count the same information, just send it once as (all, or total)